packages/babel-helper-module-transforms/src/index.ts

```diff
@@ -409,15 +417,17 @@ function buildExportInitializationStatements(
     }
   }
 
-  initStatements.push(
-    ...chunk(exportNames, 100).map(members => {
-      return buildInitStatement(
-        metadata,
-        members,
-        programPath.scope.buildUndefinedNode(),
-      );
-    }),
-  );
+  if (!noIncompleteNsImportDetection) {
```

Comment:
Nit: I think we can apply this optimization by default on any file that has 0 imports.

Reply:
Yeah we have discussions before: https://github.com/babel/babel/issues/13278#issuecomment-834397947

Reply:
Since `require` can be aliased using `const foo = require` and we have no way to know if it was so, does it make any sense just to check ES6 and dynamic imports?

Reply:
No, because `require` can still import files even if we don't track it. If we can't find a 100% safe way of checking it (and tbh I am not sure it's possible), it's probably better to avoid automatically optimizing this.